doc/rados/operations/erasure-code-clay.rst

```diff
@@ -0,0 +1,215 @@
+======================================
+Coupled LAYer code plugin
+======================================
+Coupled LAYer (CLAY) codes are erasure codes designed to save in terms of network 
+bandwidth, disk IO when a failed node/OSD/rack is being repaired. 
+
+If *jerasure* is configured with *k=8* and *m=4*, losing one OSD requires 
+reading from the eight others to repair. And recovery of say a 1GB needs
+a download of 8X 1GB = 8GB amount of information. However, in the case of
+*clay* plugin configured with *k=8* and *m=4*, when a single OSD fails
+d=11 osds are contacted and 250MB is downloaded from each of them resulting in 
+download of 11 X 250MB = 2.75GB amount of information. More general parameters are
+shown below. The benefits are huge when the repair is being done for a rack that
+stores information in the order of Tera bytes.::
+
+	S = amount of data of single OSD being repaired
+
+::
+
+	Repair Bandwidth/ Disk IO needed by jerasure, isa = k*S
+
+::
+
+	Repair Bandwidth/ Disk IO needed by clay = d*S/(d-k+1) (where k < d <= k+m-1)\
+                                                 = (k+m-1)*S/m (minimal possible for a given k,m)
+
+
+
+Erasure code profile examples
+=============================
+
+Reduced bandwidth usage can actually be observed.::
+
+        $ ceph osd erasure-code-profile set CLAYprofile \
+             plugin=lrc \
+             k=4 m=2 d=5 \
+             crush-failure-domain=host
+        $ ceph osd pool create lrcpool 12 12 erasure CLAYprofile
+
+
+Create an clay profile
+======================
+
+To create a new clay code profile::
+
+        ceph osd erasure-code-profile set {name} \
+             plugin=clay \
+             k={data-chunks} \
+             m={coding-chunks} \
+             [d={helper-chunks}] \
+             [scalar_mds={plugin-name}] \
+             [technique={technique-name}] \
+             [crush-failure-domain={bucket-type}] \
+             [directory={directory}] \
+             [--force]
+
+Where:
+
+``k={data chunks}``
+
+:Description: Each object is split in **data-chunks** parts,
+              each stored on a different OSD.
+
+:Type: Integer
+:Required: Yes.
+:Example: 4
+
+``m={coding-chunks}``
+
+:Description: Compute **coding chunks** for each object and store them
+              on different OSDs. The number of coding chunks is also
+              the number of OSDs that can be down without losing data.
+
+:Type: Integer
+:Required: Yes.
+:Example: 2
+
+``d={helper-chunks}``
+
+:Description: Number of OSDs requested to send data during recovery of
+              a single chunk. 
```

Comment:
might want 

1. note the range of possible `d`
1. explain the relation between amount data to transmit for recovery and the choosing of `d`, people might want to use minimal possible `d`. as it sounds like the less number of OSDs to contact the better, at first glance.